Error: write EPIPE when running dukat on .d.ts file

I have installed dukat using `npm install -g dukat`, Win10
Running `dukat typings.ts` in cmd outputs the following:
```js
events.js:174
      throw er; // Unhandled 'error' event
      ^

Error: write EPIPE
    at afterWriteDispatched (internal/stream_base_commons.js:78:25)
    at writeGeneric (internal/stream_base_commons.js:73:3)
    at Socket._writeGeneric (net.js:713:5)
    at Socket._write (net.js:725:8)
    at doWrite (_stream_writable.js:415:12)
    at writeOrBuffer (_stream_writable.js:399:5)
    at Socket.Writable.write (_stream_writable.js:299:11)
    at Readable.ondata (_stream_readable.js:693:20)
    at Readable.emit (events.js:198:13)
    at Readable.read (_stream_readable.js:491:10)
Emitted 'error' event at:
    at errorOrDestroy (internal/streams/destroy.js:107:12)
    at Socket.onerror (_stream_readable.js:717:7)
    at Socket.emit (events.js:203:15)
    at errorOrDestroy (internal/streams/destroy.js:107:12)
    at onwriteError (_stream_writable.js:430:5)
    at onwrite (_stream_writable.js:461:5)
    at _destroy (internal/streams/destroy.js:49:7)
    at Socket._destroy (net.js:613:3)
    at Socket.destroy (internal/streams/destroy.js:37:8)
    at afterWriteDispatched (internal/stream_base_commons.js:78:17)
```

I'm having the same problem, is there any progress or work-around?
For me this happens with pure Gradle usage on the commandline, no IntelliJ or its JVM involved.

Here a minimal reproducer:
```
$ find -type f
./build.gradle.kts
./gradle/wrapper/gradle-wrapper.jar
./gradle/wrapper/gradle-wrapper.properties
./gradle.properties
./gradlew
./gradlew.bat
./settings.gradle.kts

$ for f in gradle/wrapper/gradle-wrapper.properties gradle.properties settings.gradle.kts build.gradle.kts; do echo $f; cat $f; echo; done
gradle/wrapper/gradle-wrapper.properties
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-6.5.1-bin.zip
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists

gradle.properties
kotlin.js.experimental.generateKotlinExternals = true

settings.gradle.kts
rootProject.name = "showcase"

build.gradle.kts
plugins { kotlin("js") version "1.3.72" }
kotlin { target { nodejs() } }
repositories { mavenCentral() }
dependencies { implementation(npm("@actions/core", "1.2.4")) }

$ ./gradlew generateExternals
> Task :packageJson
> Task :kotlinNodeJsSetup SKIPPED
> Task :kotlinNpmInstall

> Task :generateExternals FAILED
events.js:187
      throw er; // Unhandled 'error' event
      ^

Error: write EPIPE
    at afterWriteDispatched (internal/stream_base_commons.js:150:25)
    at writeGeneric (internal/stream_base_commons.js:141:3)
    at Socket._writeGeneric (net.js:771:11)
    at Socket._write (net.js:783:8)
    at doWrite (_stream_writable.js:431:12)
    at writeOrBuffer (_stream_writable.js:415:5)
    at Socket.Writable.write (_stream_writable.js:305:11)
    at Readable.ondata (_stream_readable.js:727:22)
    at Readable.emit (events.js:210:5)
    at Readable.read (_stream_readable.js:525:10)
Emitted 'error' event on Socket instance at:
    at errorOrDestroy (internal/streams/destroy.js:108:12)
    at Socket.onerror (_stream_readable.js:759:7)
    at Socket.emit (events.js:215:7)
    at errorOrDestroy (internal/streams/destroy.js:108:12)
    at onwriteError (_stream_writable.js:446:5)
    at onwrite (_stream_writable.js:473:5)
    at internal/streams/destroy.js:50:7
    at Socket._destroy (net.js:664:5)
    at Socket.destroy (internal/streams/destroy.js:38:8)
    at afterWriteDispatched (internal/stream_base_commons.js:150:17) {
  errno: 'EPIPE',
  code: 'EPIPE',
  syscall: 'write'
}

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':generateExternals'.
> Process 'Generating Kotlin/JS external declarations' returns 1



* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 17s
3 actionable tasks: 3 executed

$ ./gradlew --version

------------------------------------------------------------
Gradle 6.5.1
------------------------------------------------------------

Build time:   2020-06-30 06:32:47 UTC
Revision:     66bc713f7169626a7f0134bf452abde51550ea0a

Kotlin:       1.3.72
Groovy:       2.5.11
Ant:          Apache Ant(TM) version 1.10.7 compiled on September 1 2019
JVM:          11.0.6 (AdoptOpenJDK 11.0.6+10)
OS:           Windows 10 10.0 amd64
```

I also tried with Java 8, 9 and 10, all show the same behavior

--- a/node-package/bin/dukat-cli.js
+++ b/node-package/bin/dukat-cli.js
@@ -1,6 +1,6 @@
 #!/usr/bin/env node
 
-var exec = require('child_process').exec;
+var execSync = require('child_process').execSync;
 var spawn = require('child_process').spawn;
 var path = require('path');
 var fs = require('fs');
@@ -10,7 +10,12 @@ var Readable = require('stream').Readable;
 var EventEmitter = require('events');
 
 var printError = function (errorMessage) {
-    console.error("ERROR: " + errorMessage);
+    if (typeof errorMessage !== "string") {
+        console.error("ERROR:");
+        console.error(errorMessage)
+    } else {
+        console.error("ERROR: " + errorMessage);
+    }
 };
 
 function run(command, args) {
@@ -23,18 +28,37 @@ function run(command, args) {
     return child;
 }
 
-function printNoJava() {
-    printError("It looks like you don't have java installed or it's just not reachable from command-line");
+function printInvalidJavaHome() {
+    printError("Your \"JAVA_HOME\" environment variable points to an invalid directory.");
     printError("As of now \"dukat\" requires Java Runtime Environment to be installed.");
+    printError("Either set \"JAVA_HOME\" to a proper Java installation,");
+    printError("or make sure \"java\" is callable from commandline and unset \"JAVA_HOME\".");
 }
 
-function guardJavaExists() {
-    exec("java -version", function (error, stdout, stderr) {
-        if (error) {
+function printNoJava() {
+    printError("\"java\" is not callable from commandline and no \"JAVA_HOME\" environment variable is set.");
+    printError("As of now \"dukat\" requires Java Runtime Environment to be installed.");
+    printError("Either set \"JAVA_HOME\" to a proper Java installation,");
+    printError("or make sure \"java\" is callable from commandline.");
+}
+
+function findJavaExecutable() {
+    var javaHome = process.env.JAVA_HOME
+    var result = javaHome ? path.join(javaHome, "bin", "java") : "java"
+    try {
+        execSync('"' + result + "\" -version", {
+            stdio: ['pipe', 'ignore', 'pipe']
+        });
+    } catch (error) {
+        printError(error)
+        if (javaHome) {
+            printInvalidJavaHome();
+        } else {
             printNoJava();
-            process.exit(error.code);
         }
-    });
+        process.exit(1);
+    }
+    return result
 }
 
 function createReadable() {
@@ -132,7 +156,7 @@ function cliMode(args) {
     var cliPath = path.resolve(runtimePath, "dukat-cli.jar");
     var classPath = [cliPath].join(path.delimiter);
 
-    guardJavaExists();
+    var java = findJavaExecutable();
 
     var argsProcessed = processArgs(args);
 
@@ -156,7 +180,7 @@ function cliMode(args) {
                 "-Ddukat.cli.internal.packagedir=" + packageDir,
                 "-cp", classPath, "org.jetbrains.dukat.cli.CliKt"].concat(args);
 
-            var dukatProcess = run("java", commandArgs);
+            var dukatProcess = run(java, commandArgs);
             inputStream.pipe(dukatProcess.stdin);
             return dukatProcess;
         } catch (e) {
@@ -174,7 +198,7 @@ function cliMode(args) {
             "-Ddukat.cli.internal.packagedir=" + packageDir,
             "-cp", classPath, "org.jetbrains.dukat.cli.CliKt"].concat(args);
 
-        return run("java", commandArgs);
+        return run(java, commandArgs);
     }
 
     process.exit(1);

